Source/WebDriver/soup/WebSocketServerSoup.cpp

```diff
@@ -204,10 +214,11 @@ void WebSocketServer::disconnectSession(const String& sessionId)
     RELEASE_LOG(WebDriverBiDi, "WebSockets support not implemented yet with libsoup2");
 #else
     auto connection = this->connection(sessionId);
-    if (!connection)
+    if (!connection || !connection->get())
```

Comment:
GRefPtr evaluates to false if it contains a nullptr value, so don't need to change anything here.

Reply:
Yes, but in this case, `connection` is an `std::optional`, wrapping the `GRefPtr`. So, we might have the - unexpected - case of having a non-empty `std::optional` but pointing to a null `GRefPtr`.

The reason for the optional is that while it could have been a plain `GRefPtr`, I wanted to leave the `WebSocketServer` API open for other ports that might have non-nullable `Connection` types.

FTR, in https://github.com/WebKit/WebKit/pull/42968 we're discussing hiding the `Connection` type at all from `WebSocketServer` public API.

Reply:
Ah, I didn't know it was optional, auto sometimes makes reviews more difficult